Engine/game/savegame.cpp

```diff
@@ -555,6 +555,9 @@ HSaveError DoAfterRestore(const PreservedParams &pp, const RestoredData &r_data)
     const int cf_out_chan = play.crossfading_out_channel;
     play.crossfading_in_channel = 0;
     play.crossfading_out_channel = 0;
+    
+    {
+    AudioChannelsLock lock;
     // NOTE: channels are array of MAX_SOUND_CHANNELS+1 size
     for (int i = 0; i <= MAX_SOUND_CHANNELS; ++i)
```

Comment:
Potential for setting AudioChannelsLock within the lock.
edit: you might be doing a more global lock here because you want a consistent save state.

Reply:
This is DoAfterRestore function where save data is already loaded and now applied. I am not sure if it's a good idea to make lock more global here because this function does many kinds of stuff, including *loading another room*.